packages/date-picker/src/DatePicker/DatePickerMenu/DatePickerMenuHeader/DatePickerMenuHeader.tsx

```diff
@@ -73,13 +75,28 @@ export const DatePickerMenuHeader = forwardRef<
   /**
    * Ensure that the date picker menu will not close when a select menu is open, focus is inside the select menu, and the ESC key is pressed.
    */
-  const handleEcsPress: KeyboardEventHandler<HTMLDivElement> = e => {
+  const handleMenuHeaderKeydown: KeyboardEventHandler<HTMLDivElement> = e => {
+    const { key } = e;
+    key === 'Escape' &&
+      console.log('handleMenuHeaderKeydown', {
+        key,
+        isSelectMenuOpen,
+        // @ts-ignore
+        targe: e.target.outerHTML,
+      });
+
     // `isSelectMenuOpen` provided by `PopoverProvider` is `true` if any popover _within_ the menu is open
-    if (isSelectMenuOpen) {
-      e.stopPropagation();
+    if (key === keyMap.Escape && isSelectMenuOpen) {
+      // e.stopPropagation();
+      console.log('key === keyMap.Escape && isSelectMenuOpen');
     }
   };
 
+  useEffect(() => {
```

Comment:
Not the best solution but, using useEffect to pass the value of isSelectMenuOpen to the datePickerContext so we can consume it in DatePickerComponent

Reply:
oh, onExited/onEntered could work here